### PR TITLE
story: add portaaltaak logging

### PR DESCRIPTION
**Pull request by mbritense**
for https://ritense.tpondemand.com/entity/118274-be-update-zgw-modules-to-use

### CODE REVIEW

File: zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt
Comment (Klaas-Ritense) [inline zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt:1]: Could you also add logging to the `PortaalTaakEventListener`

File: zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt [hunk 1]
```diff
@@ -47,6 +47,9 @@ import com.ritense.zakenapi.domain.rol.RolNietNatuurlijkPersoon
 import com.ritense.zakenapi.domain.rol.RolType
 import com.ritense.zakenapi.link.ZaakInstanceLinkNotFoundException
 import com.ritense.zakenapi.link.ZaakInstanceLinkService
+import mu.KLogger
+import mu.KotlinLogging
+import mu.withLoggingContext
```
Comment (Klaas-Ritense) [inline zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt]: Could you include 
`implementation project(":logging")`
And use this import instead?
```suggestion
import com.ritense.logging.withLoggingContext
```

This new `withLoggingContext` is the same but will also save the logging-context for exceptions

**Review by marijnritense — DISMISSED** [review]

File: zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt [hunk 2]
```diff
@@ -60,29 +61,46 @@ open class PortaalTaakEventListener(
...
+                    logger.debug { "Processing task with status 'ingediend' and verwerker task id '${taakObject.verwerkerTaakId}'" }
+                    val task = taskService.findTaskById(taakObject.verwerkerTaakId)
+                        ?: run {
+                            logger.warn { "Task not found with verwerker task id '${taakObject.verwerkerTaakId}'" }
+                            return
+                        }
+
+                    val receiveData = getReceiveDataActionProperty(task, it.id.id)
+                        ?: run {
+                            logger.warn { "Receive data not found for task with id '${task.id}'" }
```
Comment (marijnritense) [inline zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt]: I think you're logging this twice, once in the `getReceiveDataActionProperty` method and here. Ideally logging messages are unique. My suggestion would be to change one of the logging messages so it gives a bit more information.

File: zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt [hunk 3]
```diff
@@ -97,51 +100,63 @@ class PortaaltaakPlugin(
...
-            ?: throw IllegalStateException("Object management not found for portaal taak")
+        withLoggingContext(DelegateTask::class.java.canonicalName to delegateTask.id) {
+            logger.debug { "Creating portaaltaak for task with id '${delegateTask.id}'" }
 
-        val objectenApiPlugin = pluginService.createInstance(
-            PluginConfigurationId
-                .existingId(objectManagement.objectenApiPluginConfigurationId)
-        ) as ObjectenApiPlugin
+            val objectManagement = objectManagementService.getById(objectManagementConfigurationId)
+                ?: throw IllegalStateException("Object management not found for portaal taak")
```
Comment (marijnritense) [inline zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt]: Nitpicky thing:
```suggestion
                ?: throw IllegalStateException("Object management not found for portaaltaak")
```

File: zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt [hunk 4]
```diff
@@ -161,6 +178,9 @@ class PortaaltaakPlugin(
         )
 
         runWithoutAuthorization { taskService.complete(verwerkerTaakId) }
+
+        logger.info { "Task with id '${verwerkerTaakId}', corresponding to portaaltaak object with URL '${portaalTaakObjectUrl}', completed" }
```
Comment (marijnritense) [inline zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaaltaakPlugin.kt]:
```suggestion
        logger.info { "Task with id '${verwerkerTaakId}' for object with URL '${portaalTaakObjectUrl}' completed" }
```

**Review by marijnritense — APPROVED** [review]

**Pull request merged**